.github/workflows/generate_files.yml

```diff
@@ -1,12 +1,15 @@
 name: Generate files on merge
 
 on:
-  push:
+  pull_request_target:
+    types:
+      - closed
     branches:
       - master
 
 jobs:
   generate_file:
+    if: "github.event.pull_request.merged == true && !contains(github.event.head_commit.message, 'Add generated files')"
```

Comment:
Shouldn't the diff be empty if the commit message is "Add generated files"?
The PR action should not open a PR in this case
> If there are no changes (i.e. no diff exists with the checked-out base branch), no pull request will be created and the action exits silently.